snapcraft/parts/lifecycle.py

```diff
@@ -186,6 +186,22 @@ def _run_command(  # noqa PLR0913 # pylint: disable=too-many-branches, too-many-
         lifecycle.clean(part_names=part_names)
         return
 
+    # patchelf relys on known file list to work properly, but override-prime
+    # parts don't have the list yet, and we can't track changes in the prime.
+    if any(
+        "enable-patchelf" in _part.get("build-attributes", [])
+        for _part in getattr(project, "parts", {}).values()
+    ) and any(
+        _part.get("override-prime", None)
+        for _part in getattr(project, "parts", {}).values()
+    ):
```

Comment:
The problem should only happen if `override-prime` and `enable-patchelf` are in the same part, so we can loop over parts and test both attributes in each iteration.